### PR TITLE
[HACK][Order Creation] Ability to add quantity to the order directly via input

### LINKED ISSUE #10014: [Order Creation] Ability to add quantity to order directly rather than increase/decrease product number
From app reviews:

> It would be interesting to be able to enter the quantity of the product when creating the order, instead of increasing or decreasing just 1 unit at a time.

> product quantity adjustments require using +/- buttons. When a customer orders 56 of a single product, the inability to type a quantity is frustrating

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10014

### Description

* Enter the quantity of products in the order with a small animation
* Show the "delete" icon when the quantity is 1
"Big" quantities break things as we use float to send data, it gets serialized to exponential form and apparently backend has no idea how to handle that - https://github.com/woocommerce/woocommerce-android/issues/10328 so I restricted max direct input to 100 000, but you can still break that by using "plus" many many time

### Testing instructions

* Try both order creating and editing
* Try dark/light modes
* Try to enter and paste different input

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/f094fd66-0f25-4e2f-8b10-ba97248552e7

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by samiuelson — APPROVED** [review]
LGTM! Great work overall

💡 There is a minor issue/inconsistency – we're not able to enter amount greater than `MAX_PRODUCT_QUANTITY` manually, however, it is possible to enter a bigger amount using the "+" button. Not sure if that's expected – can be potentially misleading, but very rarely, so I think we can ignore that.

https://github.com/woocommerce/woocommerce-android/assets/4527432/2e02c3f1-184e-4812-b4e0-fcec4da9260f

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 1]
```diff
@@ -503,40 +516,123 @@ fun ExtendedProductCardContent(
...
+                )
+            }
+            BasicTextField(
+                value = textFieldValue,
+                onValueChange = { value ->
+                    try {
+                        textFieldValue =
+                            if (value.text.isNotBlank() && value.text.isNotEmpty()) {
+                                // try converting to int to validate that input is a number
+                                if (value.text.toInt() > MAX_PRODUCT_QUANTITY) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: @kidinov, we should not allow entering negative numbers. We should strip "-" char – take a look:

https://github.com/woocommerce/woocommerce-android/assets/4527432/f8de48e7-7afe-44ba-83d2-95f7cbee0a91

**Review by samiuelson — COMMENTED** [review]
I just realized it's possible to enter a negative amount which weirdly breaks up the order. I think we should strip the "-" chars from the input.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: @samiuelson, thank you! Good catch

I allow to enter only numbers between 0 and 100_000 now https://github.com/woocommerce/woocommerce-android/commit/02d2a6381d1cce9c94d1318fefb6401099137da0. Also, I ignore "plus" when 100_000 reached. Both are not nice, but we don't really have other options, without adding support of the "big" numbers in the orders

Please take a look!

**Review by samiuelson — APPROVED** [review]
Looks good!

I noticed two minor things
1. Cursor is black – brush could be set to primary color to match other text fields
2. It's possible to enter white spaces with external keyboard. This results in product being removed from the order.

**Comment by kidinov** [comment]
@samiuelson  good catch!

> Cursor is black – brush could be set to primary color to match other text fields

Changed here - 51977a1d403578ea14b4067e2a701ad4fdcde385. Tried to find how to fix that on the theme level, but no luck

> It's possible to enter white spaces with external keyboard. This results in product being removed from the order.

Yeah, that's kinda expected that it's removed. And not really expected that it would be possible to enter spaces via an external keyboard, but I think it's fine, as it is handled properly

**Pull request merged**